lib/programmemory.cpp

```diff
@@ -505,10 +507,13 @@ void ProgramMemoryState::removeModifiedVars(const Token* tok)
 {
     const ProgramMemory& pm = state;
     auto eval = [&](const Token* cond) -> std::vector<MathLib::bigint> {
-        if (conditionIsTrue(cond, pm, *settings))
-            return {1};
-        if (conditionIsFalse(cond, pm, *settings))
-            return {0};
+        MathLib::bigint result;
+        if (evaluateCondition(cond, pm, *settings, result)) {
+            if (result == 1)
+                return {1};
+            if (result == 0)
+                return {0};
+        }
```

Comment:
I think it would be better to just call execute and use the `isTrue` and `isFalse` functions:

```cpp
auto result = execute(cond, pm, *settings);
if(isTrue(result))
    return {1};
if(isFalse(result))
    return {0};
return {};
```

Reply:
Then there is no need to refactor the `evaluateCondition` function.

Reply:
And this same code could be used in `evaluateInt` as well.

Reply:
Done. As mentioned before I will not touch other code until I encounter it in a profile.